### PR TITLE
Domains: Status for transfer out

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR adds the transfer out status to the registered domain card.

This is how it appears for the domain owner:

Screenshot 2020-04-09 at 3 32 22 PM

Whereas this is how it appears for other users:

Screenshot 2020-04-09 at 3 35 49 PM

Depends on D41813-code

#### Testing instructions

- Either start a transfer out, or hack the backend to fake a transfer out status 
- Make sure the view appears as it should, for the owner and for other users
- Accept/cancel buttons should work as expected
- Assert that only the Cancel button shows when accepting the transfer is not supported

### CODE REVIEW

File: client/my-sites/domains/domain-management/components/outbound-transfer-confirmation/index.jsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+					<p>
+						<Button
+							primary
+							busy={ this.state.isAccepting }
+							disabled={ this.isRequesting() }
+							className="outbound-transfer-confirmation__accept-transfer"
+							onClick={ this.onAcceptTransferClick }
+						>
+							{ translate( 'Accept transfer' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/components/outbound-transfer-confirmation/index.jsx:146]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **29** times:
`translate( 'Accept Transfer' )` *ES Score: 11*

**Comment by hambai** [comment]
> After we merge this, can we update the link in the email to come here instead?

That might be an issue as this screen requires you to be logged in while the old screen would work for anyone. I actually started a discussion about this in p99Zz8-172-p2 since I assumed that could be a problem... :D

**Comment by fditrapani** [comment]
> I actually started a discussion about this in p99Zz8-172-p2 since I assumed that could be a problem... 

Cool, thank you! Good to know some of the background. Ideally, if we can move away from it, we should. Otherwise we'll need to make some updates to it to make it look more official.

File: client/my-sites/domains/domain-management/components/outbound-transfer-confirmation/index.jsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	renderAcceptButton() {
+		return (
+			<Button
+				primary
+				busy={ this.state.isAccepting }
+				disabled={ this.isRequesting() }
+				className="outbound-transfer-confirmation__accept-transfer"
+				onClick={ this.onAcceptTransferClick }
+			>
+				{ this.props.translate( 'Accept transfer' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/components/outbound-transfer-confirmation/index.jsx:129]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **29** times:
`translate( 'Accept Transfer' )` *ES Score: 12*

File: client/my-sites/domains/components/domain-warnings/index.jsx [hunk 3]
```diff
@@ -833,40 +832,6 @@ export class DomainWarnings extends React.PureComponent {
 		);
 	};
 
-	pendingTransfer = () => {
```
[the remark this answers is not in the document]
Comment (eltongo) [reply client/my-sites/domains/components/domain-warnings/index.jsx]: Good catch! Fixed it.

File: client/my-sites/domains/domain-management/components/outbound-transfer-confirmation/index.jsx [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+		switch ( error.error ) {
+			case 'no_pending_transfer':
+				return translate(
+					'This domain is no longer awaiting transfer. If your transfer has completed successfully or ' +
+						'you previously cancelled your domain transfer, there is nothing else you need to do.'
+				);
+			case 'domain_registration_unavailable':
+			case 'tld-in-maintenance':
+				return getMaintenanceMessageFromError( error, translate );
```
Comment (klimeryk) [inline client/my-sites/domains/domain-management/components/outbound-transfer-confirmation/index.jsx:114]: Nice job remembering about this case!

**Pull request merged**